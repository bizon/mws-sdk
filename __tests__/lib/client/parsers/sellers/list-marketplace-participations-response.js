const parseXml = require('../../../../../lib/client/parsers')
const parseListMarketplaceParticipationsResponse = require('../../../../../lib/client/parsers/sellers/list-marketplace-participations-response')

describe('lib.client.parsers.sellers.list-marketplace-participations-response', () => {
  it('should parse the ListMarketplaceParticipationsResponse example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <ListMarketplaceParticipationsResponse xmlns="https://mws.amazonservices.com/Sellers/2011-07-01">
        <ListMarketplaceParticipationsResult>
          <NextToken>MRgZW55IGNhcm5hbCBwbGVhc3VyZS6=</NextToken>
          <ListParticipations>
            <Participation>
              <MarketplaceId>ATVPDKIKX0DER</MarketplaceId>
              <SellerId>A135KKEKJAIBJ56</SellerId>
              <HasSellerSuspendedListings>No</HasSellerSuspendedListings>
            </Participation>
          </ListParticipations>
          <ListMarketplaces>
            <Marketplace>
              <MarketplaceId>ATVPDKIKX0DER</MarketplaceId>
              <Name>Amazon.com</Name>
              <DefaultCountryCode>US</DefaultCountryCode>
              <DefaultCurrencyCode>USD</DefaultCurrencyCode>
              <DefaultLanguageCode>en_US</DefaultLanguageCode>
              <DomainName>www.amazon.com</DomainName>
            </Marketplace>
          </ListMarketplaces>
        </ListMarketplaceParticipationsResult>
        <ResponseMetadata>
          <RequestId>efeab958-74e2-45d4-9018-2323084413b5</RequestId>
        </ResponseMetadata>
      </ListMarketplaceParticipationsResponse>`,
    )

    const response = parseListMarketplaceParticipationsResponse(
      '/sellers:ListMarketplaceParticipationsResponse',
      doc,
    )

    expect(response).toMatchSnapshot()
  })

  it('should parse the ListMarketplaceParticipationsByNextTokenResponse example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <ListMarketplaceParticipationsByNextTokenResponse xmlns="https://mws.amazonservices.com/Sellers/2011-07-01">
        <ListMarketplaceParticipationsByNextTokenResult>
          <NextToken>MRgZW55IGNhcm5hbCBwbGVhc3VyZS6=</NextToken>
          <ListParticipations>
            <Participation>
              <MarketplaceId>ATVPDKIKX0DER</MarketplaceId>
              <SellerId>A135KKEKWF1JAI6</SellerId>
              <HasSellerSuspendedListings>No</HasSellerSuspendedListings>
            </Participation>
          </ListParticipations>
          <ListMarketplaces>
            <Marketplace>
              <MarketplaceId>ATVPDKIKX0DER</MarketplaceId>
              <Name>Amazon.com</Name>
              <DefaultCountryCode>US</DefaultCountryCode>
              <DefaultCurrencyCode>USD</DefaultCurrencyCode>
              <DefaultLanguageCode>en_US</DefaultLanguageCode>
              <DomainName>www.amazon.com</DomainName>
            </Marketplace>
          </ListMarketplaces>
        </ListMarketplaceParticipationsByNextTokenResult>
        <ResponseMetadata>
          <RequestId>efeab958-74e2-45d4-9018-2323084413b5</RequestId>
        </ResponseMetadata>
      </ListMarketplaceParticipationsByNextTokenResponse>`,
    )

    const response = parseListMarketplaceParticipationsResponse(
      '/sellers:ListMarketplaceParticipationsByNextTokenResponse',
      doc,
      true,
    )

    expect(response).toMatchSnapshot()
  })
})
