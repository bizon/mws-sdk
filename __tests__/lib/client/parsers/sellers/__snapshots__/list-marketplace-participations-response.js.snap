// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.parsers.sellers.list-marketplace-participations-response should parse the ListMarketplaceParticipationsByNextTokenResponse example response from MWS doc 1`] = `
Object {
  "listMarketplaceParticipationsResult": Object {
    "marketplaces": Array [
      Object {
        "defaultCountryCode": "US",
        "defaultCurrencyCode": "USD",
        "defaultLanguageCode": "en_US",
        "domainName": "www.amazon.com",
        "marketplaceId": "ATVPDKIKX0DER",
        "name": "Amazon.com",
      },
    ],
    "nextToken": "MRgZW55IGNhcm5hbCBwbGVhc3VyZS6=",
    "participations": Array [
      Object {
        "hasSellerSuspendedListings": false,
        "marketplaceId": "ATVPDKIKX0DER",
        "sellerId": "A135KKEKWF1JAI6",
      },
    ],
  },
  "responseMetadata": Object {
    "requestId": "efeab958-74e2-45d4-9018-2323084413b5",
  },
}
`;

exports[`lib.client.parsers.sellers.list-marketplace-participations-response should parse the ListMarketplaceParticipationsResponse example response from MWS doc 1`] = `
Object {
  "listMarketplaceParticipationsResult": Object {
    "marketplaces": Array [
      Object {
        "defaultCountryCode": "US",
        "defaultCurrencyCode": "USD",
        "defaultLanguageCode": "en_US",
        "domainName": "www.amazon.com",
        "marketplaceId": "ATVPDKIKX0DER",
        "name": "Amazon.com",
      },
    ],
    "nextToken": "MRgZW55IGNhcm5hbCBwbGVhc3VyZS6=",
    "participations": Array [
      Object {
        "hasSellerSuspendedListings": false,
        "marketplaceId": "ATVPDKIKX0DER",
        "sellerId": "A135KKEKJAIBJ56",
      },
    ],
  },
  "responseMetadata": Object {
    "requestId": "efeab958-74e2-45d4-9018-2323084413b5",
  },
}
`;
